web/components/space-homepage/SpaceLandingPage.tsx

```diff
@@ -110,16 +114,79 @@ export function SpaceLandingPage() {
     [profileListingData?.profile_listing]
   );
 
+  
   const filteredProfileListings = useMemo(() => {
     if (searchQuery === "")
       return shuffleProfiles(allProfileListings, getDayOfYear(new Date()));
 
     const searchQueryLower = searchQuery.toLowerCase();
     const fuse = new Fuse(allProfileListings, FUSE_OPTIONS);
-
     return fuse.search(searchQueryLower).map((result) => result.item);
   }, [allProfileListings, searchQuery]);
 
+ 
+  
+  
+
+  const { userData } = useUserData();
+  const [{ data: profileData }] = useAllProfilesOfUserQuery({
+    variables: { user_id: userData?.id ?? "" },
+  });
+
+
+
+  const idsToProfileScores = useMemo(() => {
+    let tempCounter = 0;
+    const tempIdsToProfileScores = new Map();
+    
+    // loop through all profiles in directory
+    for(let i = 0; i < filteredProfileListings.length; i++){
+
+      const profileListing = filteredProfileListings[i]?.profile?.profile_listing;
+
+      if(profileListing && profileListing.profile_listing_responses){
+        for (let j = 0; j < profileListing.profile_listing_responses?.length; j++) {
+          const response = profileListing.profile_listing_responses[j];
+          if (response && response.response_html) {
+            tempCounter += response.response_html.length;
+          }
+        }
+      }
+
+      const profileHeadline = filteredProfileListings[i]?.headline;
+      if(profileHeadline && profileHeadline.length > 0){
+        
+        tempCounter += profileHeadline.length; 
+      }
+
+      tempIdsToProfileScores.set(filteredProfileListings[i].id, tempCounter);
+      tempCounter = 0;
+    }
+    return tempIdsToProfileScores;
+  }, [filteredProfileListings]);
+
+
+
+  const sortedProfileListings = filteredProfileListings.sort((a, b) =>
+    {
```

Comment:
I think the code would be more readable/maintainable if profile pictures were directly involved in the scoring algorithm.

Also, it doesn't seem like `sortedProfileListings` is used when actually rendering the profiles--were you planning to do this next?